### ISSUE TITLE
Parsing a date with invalid month values greater than 12 causes 10s to be truncated

### ISSUE THREAD

**Issue by 7qt** [wanasit/chrono#474]
The parser seems to truncate the 10s number if month is greater than 12 e.g. 13 becomes 3 and 25 becomes 5. Dates act differently if given an invalid value e.g. 29th Feb 2022 will be null. So functionality of parsing is inconsistent for day and month values

The following tests all pass (in chrono master 2.4.1), despite having invalid month values:
```
    expect(chrono.strict.parseDate('15/13/2022 11AM')).toStrictEqual(new Date(2022, 5 - 1, 13, 11))
    expect(chrono.casual.parseDate('18/13/2022 11AM')).toStrictEqual(new Date(2022, 8 - 1, 13, 11))
    expect(chrono.strict.parseDate('23/13/2022 11AM')).toStrictEqual(new Date(2022, 3 - 1, 13, 11))
```

**Comment by 7qt** [comment]
Added a pull request with failing tests that replicates the problem

**Comment by 7qt** [comment]
I've had a look into this and I think the issue is caused by the SlashDateFormatParser when it validates the given date with the regex 
`([^\\d]|^)([0-3]{0,1}[0-9]{1})[\/\.\-]([0-3]{0,1}[0-9]{1})(?:[\/\.\-]([0-9]{4}|[0-9]{2}))?(\W|$)`
An invalid one can also be validated. Due to 18/13/2022 matching the form expected. But then when this invalid date hits the validation on while loop in the executeParser method it can't extract the value from it so it removes the first digit so 18/13/22 becomes 8/13/22
![image](https://user-images.githubusercontent.com/45207596/199496010-7510ec72-7afb-4167-b725-af1be3a62abf.png)
Not sure how this would want to be fixed though as changing the regex to stop cases like this would remove the ability for this to handle both MM/DD and DD/MM format dates

**Comment by wanasit** [comment]
I think this is fixed in 49e0c63.
